ddp-workspace/projects/ddp-dsm-ui/src/app/services/dsm.service.ts

```diff
@@ -615,14 +616,21 @@ export class DSMService {
     );
   }
 
-  public downloadParticipantFile( fileName: string, bucketName: string, blob: string, realm: string, mimeType: string ): Observable<any> {
+  public getSignedUrl( ddpParticipantId: string, {fileName, bucket, blobName, guid}, realm: string):
+    Observable<any> {
     const url = this.baseUrl + DSMService.UI + 'downloadFile';
-    const map: { name: string; value: any }[] = [];
-    map.push( {name: DSMService.REALM, value: realm} );
-    map.push( {name: 'fileName', value: fileName} );
-    map.push( {name: 'bucket', value: bucketName} );
-    map.push( {name: 'blob', value: blob} );
-    return this.http.get( url, this.buildQueryBlobHeader( map )).pipe( catchError( this.handleError ) );
+    const map: { name: string; value: any }[] = [
+      {name: DSMService.REALM, value: realm} ,
+      {name: 'ddpParticipantId', value: ddpParticipantId} ,
+      {name: 'fileName', value: fileName} ,
+      {name: 'bucket', value: bucket},
+      {name: 'blobName', value: blobName},
+      {name: 'fileGuid', value: guid} ];
+    return this.http.get( url, this.buildQueryHeader( map )).pipe( catchError( this.handleError ) );
```

Comment:
If I am reading code correctly handleError just throws an error. Might as well not do the catch here at all/does not do anything for you.
(I believe that the `error:` handler in the subscription will get the error. But you have to try it! Sometimes there are surprises!)

Reply:
For the header part, as you said, we can also just pass the file values, however we have to call `this.buildQueryHeader( map )` to be able to get required HTTP headers in there as well. (For some reason GitHub doesn't let me comment on the right thread for that, sorry!)

Reply:
I think it's good if we keep this just in case we ever decide to actually do something for error handling? Currently it just throws the error so i think as you said, the `error:` part is supposed to catch it...